post_office/mail.py

```diff
@@ -270,7 +272,26 @@ def send(email):
     Email.objects.filter(id__in=email_ids).update(status=STATUS.sent)
 
     email_ids = [email.id for (email, e) in failed_emails]
-    Email.objects.filter(id__in=email_ids).update(status=STATUS.failed)
+
+    numbers_mails_get_back_in_queue = 0
+    # Retry is enabled
+    if get_max_retry() > 0:
+        emails_failed = Email.objects.filter(id__in=email_ids)
+        for email in emails_failed:
+            if email.number_of_retries < get_max_retry():
```

Comment:
We should cache this in a variable `max_retries = get_max_retries()` so we don't keep calling this function in a loop.

Reply:
ok